src/components/ContributionsEpicReminder.tsx

```diff
@@ -125,66 +87,187 @@ const errorTextStyles = css`
+                            const formValues: ReminderPayload = {
+                                email: emailAddress,
+                                reminderDate: reminderDate,
+                            };
+                            setIsSubmittingState(true);
+                            submitForm(formValues)
+                                .then(response => {
+                                    if (!response.ok) {
+                                        throw Error(response.statusText);
+                                    }
+                                    return response;
+                                })
+                                .then(response => response.json())
+                                .then(json => {
+                                    if (json !== 'OK') {
+                                        throw Error('Server error');
+                                    }
+                                    setIsSuccessState(true);
+                                })
+                                .catch((): void => setIsErrorState(true))
+                                .finally((): void => setIsSubmittingState(false));
+                        }
+                        setIsDirty(true);
+                        e.preventDefault();
+                    }}
+                >
+                    {!isSuccessState && (
+                        <>
+                            <h4 className={remindHeading}>Remind me in {reminderDateAsString}</h4>
+                            <div className={formWrapper}>
+                                <div className={inputWrapper}>
+                                    <TextInput
+                                        label="Email address"
+                                        error={inputError}
+                                        value={emailAddress}
+                                        onChange={e => setEmailAddress(e.currentTarget.value)}
+                                    />
+                                </div>
+                                <Button
+                                    iconSide="right"
+                                    icon={<SvgArrowRightStraight />}
+                                    type="submit"
+                                    disabled={isSubmittingState}
+                                    className={getCustomSubmitStyles(isSubmittingState)}
+                                >
+                                    Set a reminder
+                                </Button>
+                            </div>
+                        </>
+                    )}
+
+                    {isErrorState && (
+                        <p className={errorTextStyles}>
+                            Sorry we couldn&apos;t set a reminder for you this time. Please try
+                            again later.
+                        </p>
+                    )}
+                </form>
+
+                {!isSuccessState && (
```

Comment:
Is this an error state? Using ! against isSuccess _feels_ like it should mean failure.

Reply:
Another reason for stateMachine with a `default` state, removes this question, it's finite.